Secondary Market

yazidox opened the pull request.

yazidox committed:
add listing linked to SC

EtienneCClarke commented on hooks/useLaunchpadProgram.ts:
This file is breaking imports and should be removed.

EtienneCClarke commented on components/marketplace/listings/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                                value={newListing.swarmId}
+                                onChange={(e) => setNewListing(prev => ({ ...prev, swarmId: e.target.value }))}
+                                className="bg-white/5 border-amber-500/20"
+                                placeholder="Enter Swarm ID"
+                            />
+                        </div>
+                        <div className="grid grid-cols-2 gap-4">
+                            <div>
+                                <label className="text-sm text-white/60 mb-2 block">Number of Shares</label>
+                                <Input
+                                    type="number"
+                                    value={newListing.numberOfShares}
+                                    onChange={(e) => setNewListing(prev => ({ ...prev, numberOfShares: e.target.value }))}
+                                    className="bg-white/5 border-amber-500/20"
+                                    placeholder="Enter number of shares"
+                                />
+                            </div>
+                            <div>
+                                <label className="text-sm text-white/60 mb-2 block">Price per Share</label>
+                                <Input
+                                    type="number"
+                                    value={newListing.pricePerShare}
+                                    onChange={(e) => setNewListing(prev => ({ ...prev, pricePerShare: e.target.value }))}
+                                    className="bg-white/5 border-amber-500/20"
+                                    placeholder="Enter price in $COMPUTE"
+                                />
+                            </div>
+                        </div>
+                        <Button
+                            onClick={handleCreateListing}
+                            disabled={isLoading}
+                            className="w-full bg-amber-500/20 text-amber-300 hover:bg-amber-500/30 border border-amber-500/20"
+                        >
+                            {isLoading ? 'Creating...' : 'Create Listing'}
+                        </Button>
+                    </div>
+                ) : (
+                    <ConnectButton className="w-full" />
+                )}
+            </Card>
```
There is `<SellPositionModal />` component for this. It is incomplete but had the UI framework to expand upon.

EtienneCClarke commented on components/marketplace/listings/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                                </TableRow>
+                            ))}
+                        </TableBody>
+                    </Table>
```
Why did you not use the `<DataTable />` component for this?

EtienneCClarke commented on components/marketplace/listings/index.tsx:
```diff
@@ -0,0 +1,282 @@
+                        <Button 
+                            onClick={pagination.nextPage}
+                            disabled={!pagination.hasNextPage}
+                            className="bg-white/5 hover:bg-white/10 text-white"
```
Use text-foreground for colours, instead of using white explicitly.

Edit: This applies across the entire application.

EtienneCClarke replied:
This applies across the entire application

EtienneCClarke commented on components/market/listings/index.tsx:
Revert import change once `useLaunchpadProgram.ts` is removed.

EtienneCClarke reviewed (changes requested):
Check notes.

yazidox committed:
fix comments

EtienneCClarke commented on components/marketplace/listings/index.tsx:
```diff
@@ -0,0 +1,202 @@
+                ) : (
+                    <ConnectButton className="w-full" />
+                )}
+            </Card>
```
Have a look at the component `<SellPositionCard />` for this.

EtienneCClarke commented on hooks/useLaunchpadProgram.ts:
This can still be removed.

EtienneCClarke commented on components/market/listings/index.tsx:
Revert this import change once useLaunchpadProgram.ts is removed.

yazidox committed:
fix

EtienneCClarke committed:
Fix: Upgraded program

EtienneCClarke committed:
Added UBC logo

EtienneCClarke committed:
Added decimal resolution option to `IntlNumberFormat`

EtienneCClarke committed:
Added `find-using-pool-id` API endpoint.

EtienneCClarke committed:
Added endpoint to fetch swarms which are enabled on the secondary market.

EtienneCClarke committed:
Removed duplicate market page

EtienneCClarke committed:
Removed unused function `getSwarmUsingPoolId`

EtienneCClarke committed:
Added secondary market.

EtienneCClarke committed:
Adjusted portfolio button hover style

EtienneCClarke committed:
Removed redundant type

EtienneCClarke committed:
Updated import from `SellPositionModal` to `CreateListingModal`.
Removed disconnected modal.

EtienneCClarke committed:
Added modal to purchase listings.

EtienneCClarke committed:
Added `CancelListingModal`

EtienneCClarke committed:
Added `CreateListingModal`

EtienneCClarke committed:
Added PoolAccount type.

EtienneCClarke committed:
Added Listing and MarketListing types

EtienneCClarke committed:
Fix: `getLaunchpadProgram` returns correct type instead of generic.

EtienneCClarke committed:
Removed file.

EtienneCClarke committed:
Updated ShareListing type to match on-chain data.

EtienneCClarke committed:
removed file

EtienneCClarke committed:
removed file - duplicate functionality.

EtienneCClarke committed:
Added supported tokens.

EtienneCClarke committed:
`sortable` prop added and made optional. defaults `true`.

EtienneCClarke committed:
Added `mint`, `resolution`, `decimals` to type.

EtienneCClarke committed:
Removed file.

EtienneCClarke committed:
Added component for user listings.

EtienneCClarke committed:
Added user listing data table columns.

EtienneCClarke committed:
Removed unused imports

EtienneCClarke committed:
Removed `MarketListing` type.

EtienneCClarke committed:
Moved all data fetching to `useLaunchpadProgram`.

EtienneCClarke committed:
Added wrapping for small screens.

EtienneCClarke committed:
Refactor.

EtienneCClarke committed:
Optimised fetching swarms.

EtienneCClarke committed:
Refactor.

EtienneCClarke committed:
Refactored and made hover animation optional.

EtienneCClarke committed:
Added all program interaction.

EtienneCClarke committed:
Removed black background

EtienneCClarke committed:
Adjusted styling to markplace navigation.

EtienneCClarke committed:
Pass pool id to `SwarmRecentMarketListings` instead of swarmId

EtienneCClarke committed:
Updated columns for new MarketListing type.

EtienneCClarke committed:
Fetch listings for specific pool from `useLaunchpadProgramAccount`.
Adjusted props to take a poolid instead of `swarmId`, `numberOfListings`, and `listings`.

EtienneCClarke committed:
Added poolListings function to fetch all listings for a specific pool.

EtienneCClarke committed:
Only allow secondary market enabled swarms to be listed.

EtienneCClarke committed:
Refactor.

EtienneCClarke committed:
Adjusted min price per share to roughly $1.

EtienneCClarke committed:
Fix: build errors

EtienneCClarke committed:
Adjusted module from "CommonJS" to "ESNext"

EtienneCClarke committed:
Adjusted import

EtienneCClarke committed:
Update imports

EtienneCClarke committed:
Merge branch 'staging' into secondary-market
